server/passport/index.js

```diff
@@ -0,0 +1,95 @@
+const bcrypt = require("bcryptjs");
+const passport = require("passport");
+const LocalStrategy = require("passport-local").Strategy;
+const session = require("express-session");
+const { Retailer } = require("../models");
+
+const saltRounds = 12;
+
+passport.use(
+  "login",
+  new LocalStrategy(
+    {
+      usernameField: "username",
+      passwordField: "password",
+    },
+    (username, password, done) => {
+      return Retailer.findOne({ username: username }, "password")
+        .then(async (retailer) => {
+          if (!retailer) {
+            return done(null, false, {
+              message: "Username or password invalid.",
+            });
+          } else {
+            bcrypt
+              .compare(password, retailer.password)
```

Comment:
Like I mentioned in Register.js the retailer.password comes up as undefined because the select flag is set to false in the model.

Reply:
I can try it this way:

`}).select('+password');`

maybe that will remove the error you're experiencing.  Funny though, I never got an error when logging in.  I wonder why it's happening on yours.

Reply:
now that I'm looking at this again I may have the old code

Reply:
yup just replaced the code it works now. Sorry about that!

Reply:
I cannot reproduce this error.  It's working for me.